### DISCUSSION TITLE
Adding new language support (scheme) to tic80

### DISCUSSION THREAD

**Discussion by sthilaid** [nesbox/TIC-80#2100] [General]
Hi! I was writting my own fantasy console driven by scheme when I realized I should just add scheme support to the tic80 instead...! So I forked this project and started my implementation. I copy/pasted an existing api (squirrel) and will base off that. I already got the vendor (s7 scheme) added to the cmake and the new api file and so far so good. So a few questions:

- Are there any pitfalls I should foresee when doing this?
- Is there any interest in potentially merging my changes into the main repo, assuming it's properly done and working well.

Thanks!

**Comment by nesbox** [comment]
Hi,

Don't forget to add the `TIC_BUILD_WITH_SCHEME` definition to the `tic80_config.h`
Also, try to find all the occurencies of `TIC_BUILD_WITH_LUA` and add similar code for your lang
You changes will be merged if it's properly done and working well :)

Thanks.

**Reply by sthilaid** [reply]
Okay I made good progress, the integration is fully functionnal:

https://github.com/sthilaid/TIC-80

I still have a few pending items which I could use some help. I don't know how to generate build/assert/scheme[demo|mark].tic.dat files. It seems they are related to the default file created with the "new scheme" command. Also why the two files (demo/mark)? Is that demo supposed to match the file in the demos folder?

I also still need to add the outline support and some better error handling. Almost there ;p

Thanks for the help!

**Reply by nesbox** [reply]
> I don't know how to generate build/assert/scheme[demo|mark].tic.dat

`.dat` will be generated automatically by CMake, just put your cart to the 'demos' folder

**Reply by sthilaid** [reply]
Thanks for the tip. At this point my implementation is very functional. Everything is done and I will spend the next few days testing more thoroughly. I also only test on linux as it's my current setup. Could you have a look at the change and see if you have any comments before I initiate the pull request (I actually don't know how to do that)

easy access link: https://github.com/sthilaid/TIC-80

**Reply by nesbox** [reply]
Pls initiate the PR and we will review thew code. Don't worry, you will be able to update your code after the PR.

**Reply by bryce-carson** [reply]
> > I don't know how to generate build/assert/scheme[demo|mark].tic.dat
> 
> `.dat` will be generated automatically by CMake, just put your cart to the 'demos' folder

Only on Windows!

**Comment by ElidavididiOnGithub** [comment]
There Is Problem, The Font Does Not Support Other Languages Entirely, Try Using "PixelPixie" Its Not Out Now, But It Supports A Lotta Glyphs!

**Comment by bryce-carson** [comment]
I'm glad I found this discussion. I am adding language support for R and got stuck on this issue too. I haven't written my API yet, but I have everything else required for the `tic_script` written.
